src/plugins/borrow-plugins/plugins/aave/BorrowEngineFactory.ts

```diff
@@ -257,60 +263,159 @@ export const makeBorrowEngineFactory = (blueprint: BorrowEngineBlueprint) => {
         })
       },
       async repay(request: RepayRequest): Promise<ApprovableAction> {
-        const { nativeAmount, tokenId, fromWallet = wallet } = request
+        const { nativeAmount, fromTokenId, tokenId, fromWallet = wallet } = request
         if (zeroString(nativeAmount)) throw new Error('BorrowEngine: repay request contains no nativeAmount.')
 
-        const token = getToken(tokenId)
-        const tokenAddress = getTokenAddress(token)
-
-        const spenderAddress = (await wallet.getReceiveAddress()).publicAddress
+        const walletAddress = (await fromWallet.getReceiveAddress()).publicAddress
+
+        const debtToken = getToken(tokenId)
+        const debtTokenAddress = getTokenAddress(getToken(tokenId))
+        // HACK: Action queue doesn't wait until borrowEngine has synced before using it
+        if (instance.debts.find(debt => debt.tokenId === tokenId) == null) {
+          const reserveTokenBalances = await aaveNetwork.getReserveTokenBalances(walletAddress)
```

Comment:
should be `wallet` not `walletAddress`. Maybe rename `walletAddress` -> `fromWallet` too?